tests/test_functional.py

```diff
@@ -45,6 +45,42 @@ def test_pages_401(self, testapp, db):
         testapp.get(url_for("batch.download_batch", id=1), status=401)
 
 
+class TestErrorInterface:
+    """Test 404 pages"""
+
+    non_exist_values = [
+        5005165649,
+        5055172826,
+        0,
+        "hello",
+        "aa",
+    ]
+
+    endpoints_keys = (
+        ("detail", "rc"),
+        ("award_prep", "medal_slug"),
+    )
+
+    testcases = {}
+    for endpoint, key in endpoints_keys:
+        for value in non_exist_values:
+            testcases[f"{endpoint}-{value}"] = {
+                "endpoint": endpoint,
+                "kwargs": {key: value},
+            }
+
+    @pytest.mark.parametrize("case_name", testcases)
+    def test_non_exist_dynamic_url(self, user, testapp, case_name):
+        login(user, testapp)
+        case = self.testcases[case_name]
+        res = testapp.get(
+            url_for(f"donor.{case['endpoint']}", **case["kwargs"]), status=404
```

Comment:
The name of the endpoint should contain also the name of the blueprint rather than hardcoding the `donor.` here. In that way, you'll be able to test also `batch.batch_detail`.

Reply:
good point thank you